File: leetcode_practice/src/2500_删除每行中的最大值.rs
/*
 * @lc app=leetcode.cn id=2500 lang=rust
 *
 * [2500] 删除每行中的最大值
 */

/*
    Accepted 2023-01-31
    55/55 (4 ms)
    Your runtime beats 63.64 % of rust submissions
    Your memory usage beats 21.21 % of rust submissions (2.2 MB)
 */

// @lc code=start
impl Solution {
    pub fn delete_greatest_value(mut grid: Vec<Vec<i32>>) -> i32 {
        for v in grid.iter_mut() {
            v.sort_unstable_by(|x,y| y.cmp(x));
        }
        
        let n = grid.first().unwrap().len(); // let n = grid[0].len();
        
        let mut res = 0;
        for j in 0..n {
            res += grid.iter().map(|row| row[j]).max().unwrap();
        }
        res
    }
}
// @lc code=end

File: cpp/2500_删除每行中的最大值.cpp
/*
 * @lc app=leetcode.cn id=2500 lang=cpp
 *
 * [2500] 删除每行中的最大值
 */

#include "AllInclude.h"

// @lc code=start
/**
 * Accepted 2023-07-25
 * 55/55 (20 ms)
 * Your runtime beats 23.68 % of cpp submissions
 * Your memory usage beats 37.22 % of cpp submissions (9.2 MB)
 */
class Solution {
public:
    int deleteGreatestValue(vector<vector<int>>& grid) {
        int n = grid.at(0).size();
        int res = 0;
        for (auto& row : grid) {
            sort(row.begin(), row.end(), less<int>());
        }
        for (int j = 0; j < n; j++) {
            int m = INT_MIN;
            for (auto& row : grid) {
                m = max(m, row[j]);
            }
            res += m;
        }
        return res;
    }
};
// @lc code=end

